crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -534,32 +586,35 @@ pub fn mesh_resource_provider_system(
 
     for changed_mesh_handle in changed_meshes.iter() {
         if let Some(mesh) = meshes.get(changed_mesh_handle) {
-            let vertex_bytes = mesh
-                .get_vertex_buffer_bytes(&vertex_buffer_descriptor, true)
-                .unwrap();
-            // TODO: use a staging buffer here
-            let vertex_buffer = render_resource_context.create_buffer_with_data(
-                BufferInfo {
-                    buffer_usage: BufferUsage::VERTEX,
-                    ..Default::default()
-                },
-                &vertex_bytes,
-            );
 
-            let index_bytes = mesh.get_index_buffer_bytes().unwrap();
+            let mut index = 1; // index 0 is reserved by INDEX_BUFFER_ASSET_INDEX
```

Comment:
This *might* be out of scope for this PR, but I think we should just ditch the "index" system here and just use a text-based resource name (this ends up getting piped through to [here](https://github.com/bevyengine/bevy/blob/master/crates/bevy_wgpu/src/renderer/wgpu_render_resource_context.rs#L307-L314) where the index could just as easily be a `Cow<'static, str>` tag as part of the key to the hashmap, VertexAttribute already has a [`name` field](https://github.com/bevyengine/bevy/blob/master/crates/bevy_render/src/mesh/mesh.rs#L66) which does just that )

which will then let us look up these resources semantically when we want to bind them according to what a specific pipeline expects.

Reply:
Agree. I found the index system quite hard to work with and the hashing system sounds way more straightforward. Although I would prefere to do it in a seperate PR.